arcgis-ios-sdk-samples/Geometry/Buffer list/BufferListViewController.swift

```diff
@@ -0,0 +1,184 @@
+        let bufferPolygonFillSymbol = AGSSimpleFillSymbol(style: .solid, color: UIColor.yellow.withAlphaComponent(0.6), outline: bufferPolygonOutlineSymbol)
+        overlay.renderer = AGSSimpleRenderer(symbol: bufferPolygonFillSymbol)
+        return overlay
+    }()
+    /// An overlay to display tapped locations with red circle symbols.
+    let tapLocationsGraphicsOverlay: AGSGraphicsOverlay = {
+        let overlay = AGSGraphicsOverlay()
+        let circleSymbol = AGSSimpleMarkerSymbol(style: .circle, color: .red, size: 10)
+        overlay.renderer = AGSSimpleRenderer(symbol: circleSymbol)
+        return overlay
+    }()
+    /// An array of tapped points and buffer radii (in US feet) tuple.
+    var tappedPointsAndRadius = [(point: AGSPoint, radius: Double)]()
+    
+    /// The radius of the buffer.
+    var bufferRadius: Measurement<UnitLength> = Measurement(value: 100, unit: .miles) {
+        didSet {
+            radiusLabel.text = distanceFormatter.string(from: bufferRadius)
+        }
+    }
+    /// A formatter for the output distance string.
+    let distanceFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitStyle = .short
+        formatter.unitOptions = .naturalScale
+        formatter.numberFormatter.maximumFractionDigits = 0
+        return formatter
+    }()
+    
+    // MARK: Instance methods
+    
+    /// Creates a map.
+    ///
+    /// - Returns: A new `AGSMap` object.
+    func makeMap() -> AGSMap {
+        // The spatial reference for this sample.
+        let statePlaneNorthCentralTexas = AGSSpatialReference(wkid: 32038)!
+        let map = AGSMap(spatialReference: statePlaneNorthCentralTexas)
+        map.basemap.baseLayers.add(mapImageLayer)
+        map.initialViewpoint = AGSViewpoint(targetExtent: boundaryPolygon.extent)
+        return map
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func createButtonTapped(_ sender: UIBarButtonItem) {
+        // Clear existing buffers graphics before drawing.
+        bufferGraphicsOverlay.graphics.removeAllObjects()
+        // Create the buffers.
+        // Notice: the radius distances has the same unit of the map's spatial reference's unit.
+        if let bufferPolygon = AGSGeometryEngine.bufferGeometries(tappedPointsAndRadius.map { $0.point }, distances: tappedPointsAndRadius.map { NSNumber(value: $0.radius) }, unionResults: isUnionSwitch.isOn) {
+            let graphics = bufferPolygon.map { AGSGraphic(geometry: $0, symbol: nil) }
+            bufferGraphicsOverlay.graphics.addObjects(from: graphics)
+            setStatus(message: "Buffers created.")
+        }
+    }
+    
+    @IBAction func clearButtonTapped(_ sender: UIBarButtonItem) {
+        bufferGraphicsOverlay.graphics.removeAllObjects()
+        tapLocationsGraphicsOverlay.graphics.removeAllObjects()
+        tappedPointsAndRadius.removeAll()
+        setStatus(message: "Buffers removed. Tap on the map to add buffers.")
+    }
```

Comment:
Clear button and create button should have some UI binding

- After clear is tapped, set both to disabled
- When more than 0 buffer exists, enable both buttons